src/gui/maptools/qgsmaptoolcapture.cpp

```diff
@@ -1379,25 +1791,78 @@ void QgsMapToolCapture::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
       else if ( e->button() == Qt::RightButton )
       {
         // End of string
-        deleteTempRubberBand();
 
-        //lines: bail out if there are not at least two vertices
-        if ( mode() == CaptureLine && size() < 2 )
+        // Extract NURBS control points and weights from the rubberband
+        if ( mCurrentCaptureTechnique == Qgis::CaptureTechnique::NurbsCurve && mTempRubberBand )
         {
-          stopCapturing();
-          return;
+          const int rbPointCount = mTempRubberBand->pointsCount();
+          if ( rbPointCount > 1 )
+          {
+            // Exclude the last point (cursor position)
+            for ( int i = 0; i < rbPointCount - 1; ++i )
+            {
+              nurbsControlPoints.append( mTempRubberBand->pointFromEnd( rbPointCount - 1 - i ) );
+            }
+            // Also extract weights (in correct order)
+            const QVector<double> &rbWeights = mTempRubberBand->weights();
+            for ( int i = 0; i < rbPointCount - 1; ++i )
+            {
+              if ( i < rbWeights.size() )
+                nurbsWeights.append( rbWeights[i] );
+              else
+                nurbsWeights.append( 1.0 );
+            }
+          }
         }
 
-        //polygons: bail out if there are not at least two vertices
-        if ( mode() == CapturePolygon && size() < 3 )
+        deleteTempRubberBand();
+
+        if ( mCurrentCaptureTechnique == Qgis::CaptureTechnique::NurbsCurve )
         {
-          stopCapturing();
-          return;
+          // Minimum 4 control points required for degree 3 NURBS
+          if ( mode() == CaptureLine && nurbsControlPoints.count() < 4 )
+          {
+            stopCapturing();
+            return;
+          }
+          if ( mode() == CapturePolygon && nurbsControlPoints.count() < 4 )
+          {
+            stopCapturing();
+            return;
+          }
+        }
+        else
+        {
+          //lines: bail out if there are not at least two vertices
+          if ( mode() == CaptureLine && size() < 2 )
+          {
+            stopCapturing();
+            return;
+          }
+
+          //polygons: bail out if there are not at least two vertices
```

Comment:
assuming the code is correct
```suggestion
          //polygons: bail out if there are not at least three vertices
```